index.js

```diff
@@ -1,10 +1,10 @@
 'use strict';
 const net = require('net');
 
-const isAvailable = options => new Promise((resolve, reject) => {
+const getAvailablePort = options => new Promise((resolve, reject) => {
 	const server = net.createServer();
 	server.unref();
-	server.on('error', reject);
+	server.on('error', err => (err.code === 'EADDRINUSE') ? resolve(null) : reject(err));
```

Comment:
Well, to be frank it's perfectly normal to catch some errors and treat them as normal conditions for the purposes of some piece of code. And it's a function which tries to find available port with some settings, and returns the number, otherwise returns null.

What actually is counter-intuitive is that `EADDRINUSE` is error, not status, and is handled by separate callback.

Reply:
It all comes down to a design choice anyway. I tend to associate success to resolving a promise. Therefore, if it was not possible to retrieve the requested port, the function was unsuccessful and hence `reject` would be called. Furthermore, the caller now has to deal with `null` which, to me, from a design perspective, introduces unnecessary complexity.